extensions/exquery/restxq/src/main/java/org/exist/extensions/exquery/restxq/impl/adapters/DomEnhancingNodeProxyAdapter.java

```diff
@@ -49,103 +55,57 @@
-            }
-        };
-        
-        final Enhancer enhancer = new Enhancer();
-        enhancer.setSuperclass(NodeProxy.class);
-        enhancer.setInterfaces(clazzes);
-        enhancer.setCallbackFilter(callbackFilter);
-        enhancer.setCallbacks(callbacks);
-        
-        return (NodeProxy)enhancer.create(
-            new Class[] {
-                NodeHandle.class
-            },
-            new Object[] {
-                nodeProxy
-            });
+                .method(ElementMatchers.isDeclaredBy(domClazz).or(ElementMatchers.isDeclaredBy(Node.class)))
+                .intercept(MethodCall.invokeSelf().on(nodeProxy.getNode()).withAllArguments())
+
+                .method(ElementMatchers.isHashCode())
+                .intercept(MethodCall.invokeSelf().on(nodeProxy));
+
+        try {
+            final NodeProxy nodeProxyProxy = byteBuddyBuilder
+                    .make()
+                    .load(nodeProxy.getClass().getClassLoader())
+                    .getLoaded()
+                    .getDeclaredConstructor(Expression.class, NodeHandle.class)
+                    .newInstance(nodeProxy.getExpression(), nodeProxy);
+
+            return nodeProxyProxy;
+        } catch (final NoSuchMethodException | InstantiationException | IllegalAccessException |
+                       InvocationTargetException e) {
+            throw new IllegalStateException(e.getMessage(), e);
+        }
     }
-    
-    private static Class<? extends Node>[] getNodeClasses(final NodeProxy nodeProxy) {
-        switch(nodeProxy.getType()) {
-            
-            case Type.DOCUMENT:
-                return new Class[] {
-                    Document.class,
-                    Node.class
-                };
-            
-            case Type.ELEMENT:
-                return new Class[] {
-                    Element.class,
-                    Node.class
-                };
 
+    private static Class<? extends Node> getNodeClass(final NodeProxy nodeProxy) {
+        switch (nodeProxy.getType()) {
+            case Type.ELEMENT:
+                return Element.class;
             case Type.ATTRIBUTE:
-                return new Class[] {
-                    Attr.class,
-                    Node.class
-                };
-                
+                return Attr.class;
             case Type.TEXT:
-                return new Class[] {
-                    Text.class,
-                    Node.class
-                };
-
+                return Text.class;
+            case Type.PROCESSING_INSTRUCTION:
+                return ProcessingInstruction.class;
+            case Type.COMMENT:
+                return Comment.class;
+            case Type.DOCUMENT:
+                return Document.class;
+            case Type.CDATA_SECTION:
+                return CDATASection.class;
             default:
-                return new Class[] {
-                    Node.class
-                };
```

Comment:
Here we could use the new switch expression style, to make this more compact

Reply:
@reinhapa Personally I don't prefer the new switch expression style, however I have sent a commit to change it as requested